### PR TITLE
projects: Add "Extend Linter Integration"

### PR DESCRIPTION
**Pull request by sils**

### CODE REVIEW

**Comment by hemangsk** [comment]
ack c8123b0

**Comment by hemangsk** [comment]
unack c8123b0

File: data/projects.js [hunk 1]
```diff
@@ -79,5 +79,57 @@ support of the community guarding your back of course.",
...
+(http://api.coala.io/en/latest/Developers/Writing_Linter_Bears.html).\
+\
+This project proposal should contain a concrete list of linters that are \
+wrapped together with concrete improvement proposals to the `Lint` class \
+if needed. Because implementing a linter bear is a rather trivial task it is \
+highly recommended to spice up the project with something interesting, \
+like adding project wide linter support to the Lint class. Improving \
+integrations by adding more options can also be part of this.\
+\
+Upstream contributions to linters are encouranged, e.g. to add auto fixing \
```
Comment (hemangsk) [inline data/projects.js]: encouraged*

File: data/projects.js [hunk 2]
```diff
@@ -79,5 +79,57 @@ support of the community guarding your back of course.",
...
+		"mentors" : [
+			"AbdealiJK",
+			"Makman2",
+			"fneu",
+			"jayvdb"
+		],
+		"milestones" : {
+			"GSoC 2017 Coding Starts" : [
+				"The applicant has tried out all linters manually.",
+				"The has outlined all changes needed to the linter framework, if \
```
Comment (hemangsk) [inline data/projects.js]: The applicant*

File: index.html [hunk 3]
```diff
@@ -44,6 +44,8 @@
 
 <div ng-show="tc.isSet(1)">
 <home></home>
+<faq></faq>
+<projects></projects>
```
Comment (hemangsk) [inline index.html]: Adding projects will also add the search bar in home page, will it be ok?

File: data/projects.js [hunk 4]
```diff
@@ -79,5 +79,57 @@ support of the community guarding your back of course.",
...
+integrations by adding more options can also be part of this.\
+\
+Upstream contributions to linters are encouraged, e.g. to add auto fixing \
+abilities.",
+		"requirements" : [
+			"The applicant has added an arbitrary linter to coala.",
+			"The proposal contains a list of linters to integrate for every \
+milestone."
+		],
+		"difficulty" : "low",
```
Comment (sims1253) [inline data/projects.js]: is this really low if you suggest messing with the linter class?

File: data/projects.js [hunk 5]
```diff
@@ -79,5 +79,57 @@ support of the community guarding your back of course.",
...
+		],
+		"milestones" : {
+			"GSoC 2017 Coding Starts" : [
+				"The applicant has tried out all linters manually.",
+				"The applicant has outlined all changes needed to the linter \
+framework, if needed as a cEP."
+			],
+			"GSoC 2017 Midterm" : [
+				"The first batch of linters have been merged.",
+				"All of them are properly tested."
```
Comment (sims1253) [inline data/projects.js]: Make this a measurable goal. "At least x/y" or something.

**Comment by sils** [comment]
for now its better than nothing

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala.io/ - http://viperdev.io/ - http://gitmate.com/

File: data/projects.js [hunk 6]
```diff
@@ -79,5 +79,57 @@ support of the community guarding your back of course.",
...
+			],
+			"GSoC 2017 Final" : [
+				"The second batch of linters have been merged.",
+				"All of them are properly tested."
```
Comment (sims1253) [inline data/projects.js]: tested and documented?

File: index.html (hunk 3 above)
Comment (hemangsk) [inline index.html]: what if we add a line on main page,

"Jump To Projects" in big and bold

File: data/projects.js (hunk 5 above)
Comment (sils) [inline data/projects.js]: adding something about that to the description

**Comment by sils** [comment]
```diff
diff --git a/data/projects.js b/data/projects.js
index 6c971b2..1d28ca9 100644
--- a/data/projects.js
+++ b/data/projects.js
@@ -86,9 +86,9 @@ support of the community guarding your back of course.",
 is wrapping linters into coala. Currently we have a `Lint` class which \
 simplifies this integration. Based on this class, a large selection of \
 linters can be implemented rapidly. A tutorial on writing linter bears \
-is available at \
-[the linter documentation]\
+is available at [the linter documentation]\
 (http://api.coala.io/en/latest/Developers/Writing_Linter_Bears.html).\
+The number may vary based on the difficulty to integrate any given tool.\
 \
 This project proposal should contain a concrete list of linters that are \
 wrapped together with concrete improvement proposals to the `Lint` class \
@@ -102,9 +102,9 @@ abilities.",
                "requirements" : [
                        "The applicant has added an arbitrary linter to coala.",
                        "The proposal contains a list of linters to integrate for every \
-milestone."
+milestone, split into two batches."
                ],
-               "difficulty" : "low",
+               "difficulty" : "medium",
                "issues" : [
                        "https://github.com/coala/engagement/issues/66"
                ],
@@ -122,11 +122,13 @@ framework, if needed as a cEP."
                        ],
                        "GSoC 2017 Midterm" : [
                                "The first batch of linters have been merged.",
-                               "All of them are properly tested."
+                               "At least one improvement to the linter framework is in review state.",
+                               "All of them are properly tested and documented."
                        ],
                        "GSoC 2017 Final" : [
                                "The second batch of linters have been merged.",
-                               "All of them are properly tested."
+                               "At least one improvement to the linter framework has been merged.",
+                               "All of them are properly tested and documented."
                        ]
                },
                "initiatives" : ["GSoC"],
```

FYI

**Comment by hemangsk** [comment]
ack fbde3f1

**Comment by hemangsk** [comment]
ack 0459b78
ack bc2738b
ack fc0d8b9
ack 6daf8ea

**Comment by hemangsk** [comment]
ack bc2738b

**Comment by hemangsk** [comment]
ack fc0d8b9

**Comment by hemangsk** [comment]
ack 6daf8ea

**Comment by hemangsk** [comment]
ack 6daf8ea

**Comment by hemangsk** [comment]
ack bc2738b

**Comment by hemangsk** [comment]
ack fc0d8b9

**Pull request merged**